### PR TITLE
samples: net: echo_server/client: Decrease SRAM usage for atsamr21_xpro

### PR DESCRIPTION
**Pull request by nandojve**
In the Zephyr v3.5 to build the net samples pair echo-server/client the SRAM requirements increased for the atsamr21_xpro board when passing -DEXTRA_CONF_FILE=overlay-802154.conf argument. This adjusts the examples features to free the necessary SRAM to build and run the examples.

### CODE REVIEW

File: samples/net/sockets/echo_server/boards/atsamr21_xpro.conf [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+CONFIG_NET_BUF_RX_COUNT=5
+CONFIG_NET_BUF_TX_COUNT=5
+CONFIG_NET_MAX_CONTEXTS=2
 CONFIG_NET_MAX_CONN=1
 CONFIG_NET_MAX_ROUTES=1
 CONFIG_NET_MAX_NEXTHOPS=1
 
-CONFIG_SHELL_STACK_SIZE=768
-CONFIG_SHELL_CMD_BUFF_SIZE=64
+CONFIG_SHELL_STACK_SIZE=384
```
Comment (jukkar) [inline samples/net/sockets/echo_server/boards/atsamr21_xpro.conf:20]: This looks really low but if you have tested that things work ok, then no worries.

File: samples/net/sockets/echo_server/boards/atsamr21_xpro.conf [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+CONFIG_SHELL_CMD_BUFF_SIZE=16
 CONFIG_SHELL_ARGC_MAX=6
-CONFIG_SHELL_HISTORY_BUFFER=64
+CONFIG_SHELL_HISTORY_BUFFER=4
+
```
Comment (jukkar) [inline samples/net/sockets/echo_server/boards/atsamr21_xpro.conf:24]: Following things could be disabled without too much problems:
```
CONFIG_NET_SHELL_DYN_CMD_COMPLETION=n
CONFIG_NET_INTERFACE_NAME=n
CONFIG_NET_CONTEXT_REUSEADDR=n
CONFIG_NET_CONTEXT_REUSEPORT=n
```
The net-shell dynamic cmd completion is nice helper but can be safely turned off.
The network interface name was just introduced but for echo-* apps it is not needed.
Depending on sample app needs, the two reuse options are optional and can be disabled.

File: samples/net/sockets/echo_server/boards/atsamr21_xpro.conf [hunk 3]
```diff
@@ -1,23 +1,31 @@
...
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_NET_IP_DSCP_ECN=n
+CONFIG_NET_STATISTICS=n
+CONFIG_NET_MGMT_EVENT_STACK_SIZE=384
```
Comment (jukkar) [inline samples/net/sockets/echo_server/boards/atsamr21_xpro.conf:28]: This looks too very low, but might work ok. Note that there is `CONFIG_NET_DEBUG_MGMT_EVENT_STACK` option which prints stack usage after each event, and that can be used to verify the numbers.

**Review by jukkar — DISMISSED** [review]

File: samples/net/sockets/echo_server/boards/atsamr21_xpro.conf (hunk 1 above)
Comment (nandojve) [reply samples/net/sockets/echo_server/boards/atsamr21_xpro.conf:20]: I'm getting feedback from others that need terminal. I'll return on this soon.

File: samples/net/sockets/echo_client/boards/atsamr21_xpro.conf [hunk 4]
```diff
@@ -1,23 +1,31 @@
...
+CONFIG_NET_BUF_TX_COUNT=5
+CONFIG_NET_MAX_CONTEXTS=2
 CONFIG_NET_MAX_CONN=1
 CONFIG_NET_MAX_ROUTES=1
 CONFIG_NET_MAX_NEXTHOPS=1
 
-CONFIG_SHELL_STACK_SIZE=768
-CONFIG_SHELL_CMD_BUFF_SIZE=64
+CONFIG_SHELL_STACK_SIZE=384
+CONFIG_SHELL_CMD_BUFF_SIZE=16
```
Comment (MrKevinWeiss) [inline samples/net/sockets/echo_client/boards/atsamr21_xpro.conf:21]: With this limitation commands such as `ieee802154 get_pan_id` cannot be issued.

I tested with the following and was able to perform [interop testing with RIOT](https://github.com/RIOT-OS/Release-Specs/tree/master/08-interop#task-11---udp-exchange-between-iotlab-m3-and-zephyr).  Maybe `CONFIG_SHELL_STACK_SIZE` can be reduced but this compiles and passes.
```suggestion
CONFIG_SHELL_STACK_SIZE=512
CONFIG_SHELL_CMD_BUFF_SIZE=32
```

**Review by MrKevinWeiss — COMMENTED** [review]
This fixes our problem!  It would be nice to see it in (after the suggestion is applied)!

**Comment by MaureenHelm** [comment]
@nandojve will you return to this before the 3.6 merge window closes this week?

**Comment by nandojve** [comment]
Yes, thank you!

**Comment by MrKevinWeiss** [comment]
It would be nice to have in from my side :)

File: samples/net/sockets/echo_server/boards/atsamr21_xpro.conf (hunk 2 above)
Comment (nandojve) [reply samples/net/sockets/echo_server/boards/atsamr21_xpro.conf:24]: Applied NET_CONTEXT_REUSE

File: samples/net/sockets/echo_server/boards/atsamr21_xpro.conf (hunk 3 above)
Comment (nandojve) [reply samples/net/sockets/echo_server/boards/atsamr21_xpro.conf]:
```
*** Booting Zephyr OS build zephyr-v3.5.0-5222-g03772eb0d806 ***


uart:~$ I: Initializing network
I: IPv6 address: 2001:db8::1
I: Run echo server
I: Network connected
I: Starting...
I: Waiting for UDP packets on port 4242 (IPv6)...
D: (udp6): UDP (IPv6): Received and replied with 21 bytes
D: (udp6): UDP (IPv6): Received and replied with 46 bytes
D: (udp6): UDP (IPv6): Received and replied with 12 bytes
D: (udp6): UDP (IPv6): Received and replied with 54 bytes
D: (udp6): UDP (IPv6): Received and replied with 84 bytes
D: (udp6): UDP (IPv6): Received and replied with 46 bytes
D: (udp6): UDP (IPv6): Received and replied with 65 bytes
D: (udp6): UDP (IPv6): Received and replied with 3 bytes
D: (udp6): UDP (IPv6): Received and replied with 116 bytes
D: (udp6): UDP (IPv6): Received and replied with 24 bytes
D: (udp6): UDP (IPv6): Received and replied with 95 bytes
D: (udp6): UDP (IPv6): Received and replied with 25 bytes
D: (udp6): UDP (IPv6): Received and replied with 67 bytes
D: (udp6): UDP (IPv6): Received and replied with 97 bytes
D: (udp6): UDP (IPv6): Received and replied with 108 bytes
```

```
*** Booting Zephyr OS build zephyr-v3.5.0-5222-g03772eb0d806 ***


uart:~$ I: Initializing network
I: IPv6 address: 2001:db8::2
I: Run echo client
I: Network connected
I: Starting...
D: IPv6 UDP: Sent 21 bytes
D: IPv6 UDP: Received and compared 21 bytes, all ok
D: IPv6 UDP: Sent 46 bytes
D: IPv6 UDP: Received and compared 46 bytes, all ok
D: IPv6 UDP: Sent 12 bytes
D: IPv6 UDP: Received and compared 12 bytes, all ok
D: IPv6 UDP: Sent 54 bytes
D: IPv6 UDP: Received and compared 54 bytes, all ok
D: IPv6 UDP: Sent 84 bytes
D: IPv6 UDP: Received and compared 84 bytes, all ok
D: IPv6 UDP: Sent 46 bytes
D: IPv6 UDP: Received and compared 46 bytes, all ok
D: IPv6 UDP: Sent 65 bytes
D: IPv6 UDP: Received and compared 65 bytes, all ok
D: IPv6 UDP: Sent 3 bytes
D: IPv6 UDP: Received and compared 3 bytes, all ok
D: IPv6 UDP: Sent 116 bytes
D: IPv6 UDP: Received and compared 116 bytes, all ok
D: IPv6 UDP: Sent 24 bytes
D: IPv6 UDP: Received and compared 24 bytes, all ok
D: IPv6 UDP: Sent 95 bytes
D: IPv6 UDP: Received and compared 95 bytes, all ok
D: IPv6 UDP: Sent 25 bytes
D: IPv6 UDP: Received and compared 25 bytes, all ok
D: IPv6 UDP: Sent 67 bytes
D: IPv6 UDP: Received and compared 67 bytes, all ok
D: IPv6 UDP: Sent 97 bytes
D: IPv6 UDP: Received and compared 97 bytes, all ok
D: IPv6 UDP: Sent 108 bytes
D: IPv6 UDP: Received and compared 108 bytes, all ok
```

File: samples/net/sockets/echo_client/boards/atsamr21_xpro.conf (hunk 4 above)
Comment (nandojve) [reply samples/net/sockets/echo_client/boards/atsamr21_xpro.conf]: Rollback `SHELL_STACK_SIZE=768` and applied `SHELL_CMD_BUFF_SIZE=32` as suggested

**Comment by nandojve** [comment]
Hi @MaureenHelm ,

This seems to be ready from my side. I address all comments in regards to best values and made a test on HW yesterday. So, it only depends on approvals.

Thank you again to bring this to my attention : )

**Review by rlubos — APPROVED** [review]

**Review by MrKevinWeiss — APPROVED** [review]
Looks good!

**Review by jukkar — APPROVED** [review]

**Pull request merged**